src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -5012,11 +5193,22 @@ public C(object b)
 }
 ";
             var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe);
+#if ALLOW_IN_CONSTRUCTOR_INITIALIZER
```

Comment:
I didn't understand the purpose of the `ALLOW_IN_CONSTRUCTOR_INITIALIZER` pragma. Why not just update the test code?